agent/rpc/middleware/interceptors.go

```diff
@@ -157,3 +159,19 @@ func GetNetRPCInterceptor(recorder *RequestRecorder) rpc.ServerServiceCallInterc
 		recorder.Record(reqServiceMethod, RPCTypeNetRPC, reqStart, argv.Interface(), err != nil)
 	}
 }
+
+func GetNetRPCRateLimitingInterceptor(requestLimitsHandler rpcRate.RequestLimitsHandler) rpc.PreBodyInterceptor {
+
+	return func(reqServiceMethod string, sourceAddr net.Addr) error {
+		op := rpcRate.Operation{
+			Name:       reqServiceMethod,
+			SourceAddr: sourceAddr,
+			Type:       rpcRateLimitSpecs[reqServiceMethod],
+		}
+
+		// net/rpc does not provide a way to encode the nuances of the
+		// error response (retry or retry elsewhere) so the error string
+		// from the rate limiter is all that we have.
+		return requestLimitsHandler.Allow(op)
```

Comment:
Not specific to this PR but we need to make sure that the error returned here have a string that is unique enough to be able to reason about on the client side. As for net/RPC (unlike gRPC) the string error is returned as is in the wire and there is no concepts of error codes and retryable errors.So we need to do the check on the client side based on the string.

Nothing actionable in my comment, just some thinking 😅

Reply:
Yup, the error string is one of these:
```Go 
	ErrRetryElsewhere = errors.New("rate limit exceeded, try a different server")
	ErrRetryLater = errors.New("rate limit exceeded, try again later")
```